io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -1008,25 +1016,25 @@ impl LvsLvol for Lvol {
         }
     }
 
-    /// Build Snapshot Parameters from Blob.
-    fn build_snapshot_param(&self, _blob: *mut spdk_blob) -> SnapshotParams {
-        // TODO: need to Integrate with Snapshot Property Enumeration
-        // Currently it is stub.
-        SnapshotParams::new(
-            Some(self.name()),
-            Some(self.name()),
-            Some(self.name()),
-            Some(self.name()),
-            Some(self.name()),
-            Some(Utc::now().to_string()),
-            false,
-        )
+    /// Get the parent spdk_blob from the current blob.
+    unsafe fn bs_iter_parent(
+        &self,
+        curr_blob: *mut spdk_blob,
+    ) -> Option<*mut spdk_blob> {
+        let parent_blob = spdk_bs_get_parent_blob(curr_blob);
+        if parent_blob.is_null() {
+            None
+        } else {
+            Some(parent_blob)
+        }
     }
+
     /// Wrapper function to destroy replica and its associated snapshot if
     /// replica is identified as last clone.
     async fn destroy_replica(mut self) -> Result<String, Error> {
         let snapshot_lvol = self.is_snapshot_clone();
         let name = self.name();
+        self.reset_snapshot_tree_usage_cache(true);
```

Comment:
Do we need to reset when destroying a replica?

Reply:
Yes. This is the common code for destroying clone. When clone destroyed for the snapshot, then complete chain to be refreshed.